pkg/generate/app/pipeline.go

```diff
@@ -38,20 +39,20 @@ func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
 	}, nil
 }
 
-// NewBuildPipeline creates a new pipeline with components that are
-// expected to be built
+// NewBuildPipeline creates a new pipeline with components that are expected to
+// be built.
 func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy *BuildStrategyRef, env Environment, source *SourceRef) (*Pipeline, error) {
 	name, ok := NameSuggestions{source, input}.SuggestName()
 	if !ok {
-		name = fmt.Sprintf("app%d", rand.Intn(10000))
+		return nil, ErrNameRequired
 	}
+	name = namer.LimitLength(name, v2.RepositoryNameTotalLengthMax-kutil.DNS1123LabelMaxLength-1)
```

Comment:
> Do we know why that line was there and who added it?

I added it in a previous PR that was reverted, and considered adding here again, but on second thought I removed it from this PR.